src/Nethermind/Nethermind.Blockchain/FullPruning/FullPruner.cs

```diff
@@ -71,6 +71,11 @@ public class FullPruner : IDisposable
             _pruningTrigger.Prune += OnPrune;
             _logger = _logManager.GetClassLogger();
             _minimumPruningDelay = TimeSpan.FromHours(_pruningConfig.FullPruningMinimumDelayHours);
+
+            if (_pruningConfig.ShutdownAfterFullPrune)
+            {
+                _fullPruningDb.PruningFinished += HandlePruningFinished;
```

Comment:
For my use case, I would like it to restart regardless of success or failure because I don't believe it should be in `Pruning.Mode Full` unless it's actively pruning. If it fails, we want to set it back to `Memory` mode while the user diagnoses and is ready to try again.

A possible solution would be to make this an enum instead of a bool, with options like
- `FullPruningCompletionBehavor.None` (default)
- `FullPruningCompletionBehavor.AlwaysShutdown` (what I need)
- `FullPruningCompletionBehavor.ShutdownOnSuccess`

Reply:
Ok, I have made the above changes. The setting name has been changed from `ShutdownAfterFullPrune` to `FullPruningCompletionBehavor`, which is an enum. The default is `FullPruningCompletionBehavor.None` so it doesn't break existing deployments.